components/Billing/Billing.tsx

```diff
@@ -222,8 +224,15 @@ class BillingImpl extends JCComponent<Props, State> {
         },
         authMode: GRAPHQL_AUTH_MODE.AMAZON_COGNITO_USER_POOLS,
       })
-      console.log({ invoice: invoice.data.previewInvoice?.invoice })
-      this.setState({ invoice: invoice.data.previewInvoice?.invoice })
+      this.setState({ invoiceQueue: [...this.state.invoiceQueue, newInvoice] }, async () => {
+        const currentIndex = this.state.invoiceQueue.length - 1
+        const invoice = await this.state.invoiceQueue[currentIndex]
+
+        if (currentIndex === this.state.invoiceQueue.length - 1) {
+          console.log({ invoice: invoice.data.previewInvoice?.invoice })
+          this.setState({ invoice: invoice.data.previewInvoice?.invoice })
+        }
+      })
```

Comment:
if the length of invoiceQueue is changed while waiting for the promise to resolve, it is ignored.